samples/wifi/raw_tx_packet/src/wifi_connection.c

```diff
@@ -0,0 +1,261 @@
+	case NET_EVENT_WIFI_CONNECT_RESULT:
+		handle_wifi_connect_result(cb);
+		break;
+	case NET_EVENT_WIFI_DISCONNECT_RESULT:
+		handle_wifi_disconnect_result(cb);
+		break;
+	default:
+		break;
+	}
+}
+
+static void print_dhcp_ip(struct net_mgmt_event_callback *cb)
+{
+	/* Get DHCP info from struct net_if_dhcpv4 and print */
+	const struct net_if_dhcpv4 *dhcpv4 = cb->info;
+	const struct in_addr *addr = &dhcpv4->requested_ip;
+	char dhcp_info[128];
+
+	net_addr_ntop(AF_INET, addr, dhcp_info, sizeof(dhcp_info));
+
+	LOG_INF("DHCP IP address: %s", dhcp_info);
+}
+
+static void net_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+				    uint32_t mgmt_event, struct net_if *iface)
+{
+	switch (mgmt_event) {
+	case NET_EVENT_IPV4_DHCP_BOUND:
+		print_dhcp_ip(cb);
+		break;
+	default:
+		break;
+	}
+}
+
+static int __wifi_args_to_params(struct wifi_connect_req_params *params)
+{
+	params->timeout = SYS_FOREVER_MS;
+
+	/* SSID */
+	params->ssid = CONFIG_RAW_TX_PACKET_SAMPLE_SSID;
+	params->ssid_length = strlen(params->ssid);
+
+#if defined(CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_WPA2)
+	params->security = 1;
+#elif defined(CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_WPA2_256)
+	params->security = 2;
+#elif defined(CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_WPA3)
+	params->security = 3;
+#else
+	params->security = 0;
+#endif
+
+#if !defined(CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_NONE)
+	params->psk = CONFIG_RAW_TX_PACKET_SAMPLE_PASSWORD;
+	params->psk_length = strlen(params->psk);
+#endif
+	params->channel = WIFI_CHANNEL_ANY;
+
+	/* MFP (optional) */
+	params->mfp = WIFI_MFP_OPTIONAL;
+
+	return 0;
+}
+
+static int wifi_connect(void)
+{
+	struct net_if *iface = NULL;
+	static struct wifi_connect_req_params cnx_params;
+
+	iface = net_if_get_first_wifi();
+	if (!iface) {
+		LOG_ERR("Failed to get Wi-Fi iface");
+		return -1;
+	}
+
+	context.connected = false;
+	context.connect_result = false;
+	__wifi_args_to_params(&cnx_params);
+
+	if (net_mgmt(NET_REQUEST_WIFI_CONNECT, iface,
+		     &cnx_params, sizeof(struct wifi_connect_req_params))) {
+		LOG_ERR("Connection request failed");
+
+		return -ENOEXEC;
+	}
+
+	LOG_INF("Connection requested");
+
+	return 0;
+}
+
+int try_wifi_connect(void)
+{
+	memset(&context, 0, sizeof(context));
+
+	net_mgmt_init_event_callback(&wifi_shell_mgmt_cb,
+				     wifi_mgmt_event_handler,
+				     WIFI_SHELL_MGMT_EVENTS);
+
+	net_mgmt_add_event_callback(&wifi_shell_mgmt_cb);
+
+	net_mgmt_init_event_callback(&net_shell_mgmt_cb,
+				     net_mgmt_event_handler,
+				     NET_EVENT_IPV4_DHCP_BOUND);
+
+	net_mgmt_add_event_callback(&net_shell_mgmt_cb);
+
+	LOG_INF("Static IP address (overridable): %s/%s -> %s",
+		CONFIG_NET_CONFIG_MY_IPV4_ADDR,
+		CONFIG_NET_CONFIG_MY_IPV4_NETMASK,
+		CONFIG_NET_CONFIG_MY_IPV4_GW);
+
+	while (1) {
+		wifi_connect();
+
+		while (!context.connect_result) {
+			cmd_wifi_status();
+			k_sleep(K_MSEC(STATUS_POLLING_MS));
+		}
+
+		if (context.connected) {
+			break;
+		} else if (!context.connect_result) {
```

Comment:
It now has a enum for `connect_result`, so, log the result. (We should do this for existing samples as well).

Reply:
I didn't find enum definition for ``connect_result``. Can you point me to that file?

Reply:
I didn't find enum definition for ``connect_result``. Can you point me to that file?

Reply:
https://github.com/nrfconnect/sdk-zephyr/blob/main/include/zephyr/net/wifi_mgmt.h#L339